staging/src/k8s.io/client-go/openapi/openapitest/fake.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapitest
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/client-go/openapi"
+)
+
+// NewFileClient returns a pointer to a testing
+// FileOpenAPIClient, which parses the OpenAPI GroupVersion
+// files in the passed directory path. Example GroupVersion
+// OpenAPI filename for apps/v1 would look like:
+//
+//	apis__apps__v1_openapi.json
+//
+// The current directory housing these hard-coded GroupVersion
+// OpenAPI V3 specification files is:
+//
+//	<K8S_ROOT>/api/openapi-spec/v3
+//
+// An example to invoke this function for the test files:
+//
+//	NewFileClient("../../../api/openapi-spec/v3")
+//
+// This function will search all subdirectories for .json files.
+func NewFileClient(fullDirPath string) openapi.Client {
+	_, err := os.Stat(fullDirPath)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to find Open API test file directory: %s\n", fullDirPath))
+	}
+	values := map[string]openapi.GroupVersion{}
+	err = filepath.Walk(fullDirPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		filename := info.Name()
+		if !info.IsDir() && (filepath.Ext(filename) == ".json") {
+			apiFilename, err := apiFilepath(filename)
+			if err != nil {
+				// continue if filename is not required format
+				return nil
```

Comment:
Silencing the error, do you have examples already that don't pass the test here? If I'm reading carefully, all files in that dir: https://github.com/kubernetes/kubernetes/tree/master/api/openapi-spec/v3 finish with `_openapi.json`.

Reply:
Fixed. Also added comments documenting panic when file does not parse. Please have another look.